File: lightbulb-dialogs/src/main/java/com/github/RooneyAndShadows/lightbulb/dialogs/base/constraints/bottomsheet/BottomSheetDialogConstraintsBuilder.java
package com.github.RooneyAndShadows.lightbulb.dialogs.base.constraints.bottomsheet;

import com.github.RooneyAndShadows.lightbulb.dialogs.base.BaseDialogFragment;
import com.github.RooneyAndShadows.lightbulb.dialogs.base.constraints.base.BaseDialogConstraintBuilder;

@SuppressWarnings("FieldCanBeLocal")
public class BottomSheetDialogConstraintsBuilder extends BaseDialogConstraintBuilder {
    private final int DEF_MAX_HEIGHT_PERCENTS = 65;
    private final int DEF_MIN_HEIGHT_PERCENTS = 10;
    private int minHeight;
    private int maxHeight;

    public BottomSheetDialogConstraintsBuilder(BaseDialogFragment dialogFragment) {
        super(dialogFragment);
    }

    public BottomSheetDialogConstraintsBuilder Default() {
        int orientation = dialogFragment.getResources().getConfiguration().orientation;
        this.minHeight = getPercentOfWindowHeight(DEF_MIN_HEIGHT_PERCENTS);
        this.maxHeight = getPercentOfWindowHeight(DEF_MAX_HEIGHT_PERCENTS);
        return this;
    }

    public BottomSheetDialogConstraintsBuilder withMinHeight(int minHeight) {
        this.minHeight = minHeight;
        return this;
    }

    public BottomSheetDialogConstraintsBuilder withMaxHeight(int maxHeight) {
        this.maxHeight = maxHeight;
        return this;
    }

    public BottomSheetDialogConstraintsBuilder withMinHeightInPercents(int minHeightPercent) {
        minHeightPercent = validatePercentNumber(minHeightPercent);
        minHeight = getPercentOfWindowHeight(minHeightPercent);
        return this;
    }

    public BottomSheetDialogConstraintsBuilder withMaxHeightInPercents(int maxHeightPercent) {
        maxHeightPercent = validatePercentNumber(maxHeightPercent);
        maxHeight = getPercentOfWindowHeight(maxHeightPercent);
        return this;
    }

    public BottomSheetDialogConstraints build() {
        return new BottomSheetDialogConstraints(minHeight, maxHeight);
    }
}

File: lightbulb-dialogs/src/main/java/com/github/RooneyAndShadows/lightbulb/dialogs/base/constraints/bottomsheet/BottomSheetDialogConstraintsBuilder.kt
package com.github.rooneyandshadows.lightbulb.dialogs.base.constraints.bottomsheet

import com.github.rooneyandshadows.lightbulb.dialogs.base.BaseDialogFragment
import com.github.rooneyandshadows.lightbulb.dialogs.base.constraints.base.BaseDialogConstraintBuilder

class BottomSheetDialogConstraintsBuilder(dialogFragment: BaseDialogFragment) : BaseDialogConstraintBuilder(dialogFragment) {
    private val defaultMaxHeightPercents = 65
    private val defaultMinHeightPercents = 10
    private var minHeight = 0
    private var maxHeight = 0

    fun default(): BottomSheetDialogConstraintsBuilder {
        val orientation = dialogFragment.resources.configuration.orientation
        minHeight = getPercentOfWindowHeight(defaultMinHeightPercents)
        maxHeight = getPercentOfWindowHeight(defaultMaxHeightPercents)
        return this
    }

    fun withMinHeight(minHeight: Int): BottomSheetDialogConstraintsBuilder {
        this.minHeight = minHeight
        return this
    }

    fun withMaxHeight(maxHeight: Int): BottomSheetDialogConstraintsBuilder {
        this.maxHeight = maxHeight
        return this
    }

    fun withMinHeightInPercents(minHeightPercent: Int): BottomSheetDialogConstraintsBuilder {
        minHeight = getPercentOfWindowHeight(validatePercentNumber(minHeightPercent))
        return this
    }

    fun withMaxHeightInPercents(maxHeightPercent: Int): BottomSheetDialogConstraintsBuilder {
        maxHeight = getPercentOfWindowHeight(validatePercentNumber(maxHeightPercent))
        return this
    }

    fun build(): BottomSheetDialogConstraints {
        return BottomSheetDialogConstraints(minHeight, maxHeight)
    }
}